README.md

```diff
@@ -136,122 +171,280 @@ Every resource in this repository is essential for the overall template setup. N
+This Template contains an example implementation of all features includes in the Generator, both in React and in Nunjucks.
 
-### Template Specific Filters
+### React
 
-Check out [filters/mermaidDiagram.js](filters/mermaidDiagram.js) file to see an example of the custom filter that template developers can write for their template files. It exports a JavaScript function called `generateMermaidDiagram` that accepts parsed AsyncAPI document as an argument and returns a [Mermaid class diagram](https://mermaid-js.github.io/mermaid/#/) that visualizes the data model basing on schemas provided in the document.
+The list of resources that are relevant for this Template using React:
 
-Important things to notice:
-- The file contains multiple functions, but only one is exported. It means you can also have private functions and use them in your filter function to better structure your code.
-- It is good to properly validate the input passed to the filter and throw a clear error.  
-  ```js
-   if (!asyncapi || typeof asyncapi.version !== 'function') throw new Error('You need to pass entire parsed AsyncAPI document as an argument. Try this "{{ asyncapi | generateMermaidDiagram }}"');
-  ```
-  As a result, you get a helpful error during template development like:
-  ```js
-  Error: You need to pass the AsyncAPI document as an argument. Try this "{{ asyncapi | generateMermaidDiagram }}"
-  ```
-  Instead of 
-  ```js
-  TypeError: asyncapi.hasChannels is not a function
-  ```
-  It is crucial because, by default, errors that are generated from Nunjucks do not tell you where exactly you have an error in your filter. You can try to use `--debug` flag when running generation during development, but it is helpful only in some instances.
-- Parsed AsyncAPI document that you can access in the filter comes from the AsyncAPI JavaScript Parser. It provides you with extensive [API](https://github.com/asyncapi/parser-js/blob/master/API.md) that you can use to navigate through the document to extract certain parts from it, like for example:
-  ```js
-    const OPERATIONS = ['publish', 'subscribe'];
+- `template` is a directory where you keep all the files that will be processed by the Generator using React. Only files with the `.js`, `.jsx` and `.cjs` extensions are taken and processed by the Generator. The rest are skipped.
+- `partials` is a directory where you keep reusable parts (called also components) of the templates. You can also put there helper functions to handle perform actions on AsyncAPI data inside components.
 
-    asyncapi.channelNames().forEach(channelName => {
-      const channel = asyncapi.channel(channelName);
+> **NOTE**: `partials` directory is recommended place to split reusable chunks/helpers. The reusable parts can be located both in the `template` folder as in another named folder. The only exception is the `hooks` folder, it is reserved for the Generator.
```

Comment:
Is that actually true? If I put a component inside `template`, it would also be generated as a separate file, right?

Reply:
Only if certain criteria are fulfilled i.e. those stated in https://github.com/asyncapi/template-for-generator-templates/blob/c0c984e385ca2dd78fba27cf4fc511a4f42ae747/README.md#file-component. Otherwise it discard the rendering of the file (unless if you specify it as non-rendering ofcourse). This means that in practise you can actually place reuseable components inside the `template` directory 😄